### PR TITLE
fix(auth): OauthService wait for initial restoreState

### PR DESCRIPTION
**Pull request by aleespryker**
closes: [HRZ-89566](https://spryker.atlassian.net/browse/HRZ-89566)

[HRZ-89566]: https://spryker.atlassian.net/browse/HRZ-89566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/auth/src/oauth/oauth.service.ts [hunk 1]
```diff
@@ -192,11 +203,20 @@ export class OauthService implements AuthService, AuthTokenService {
     return this.state$;
   }
 
-  protected restoreState(): void {
-    this.storageService
+  protected initState(): Observable<OauthServiceState | null> {
```
Comment (tolerants) [inline libs/platform/auth/src/oauth/oauth.service.ts]: Shoudn't we update tests?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: libs/platform/auth/src/oauth/oauth.service.ts [hunk 2]
```diff
@@ -192,11 +203,20 @@ export class OauthService implements AuthService, AuthTokenService {
...
+      .pipe(
+        distinctUntilChanged(),
+        map((state) => {
+          this.state$.next({ ...state });
+          return state;
+        })
+      );
+  }
+
+  protected restoreState(): void {
```
Comment (nsilva-spryker) [inline libs/platform/auth/src/oauth/oauth.service.ts]: Do we still need it and why?

**Review by nsilva-spryker — CHANGES_REQUESTED** [review]

File: libs/platform/auth/src/oauth/oauth.service.ts (hunk 2 above)
Comment (aleespryker) [reply libs/platform/auth/src/oauth/oauth.service.ts]: It's still being used by `invokeStoredToken`, but I can move the implementation of `restoreState` there so we won't need it anymore.

Comment (nsilva-spryker) [reply libs/platform/auth/src/oauth/oauth.service.ts]: yes, let's do this small refactor and move implementation to `invokeStoredToken`

**Review by nsilva-spryker — DISMISSED** [review]

**Review by tarashaliuk — APPROVED** [review]

**Review by tolerants — APPROVED** [review]

**Review by nsilva-spryker — APPROVED** [review]

**Pull request merged**